Adding some explanations and removing spelling errors (#253)
nf-core/differentialabundance: docs/usage.md

## Introduction

> Differential analysis is a frequent task in a variety of use cases. In essence, all these use cases entail taking an input matrix containing features (e.g. genes) and observations (e.g. samples), and comparing groups of observations in all or a subset of the features. The feature/ observation language here reflects our hope that this workflow will extend in future to encompass a variety of applications where an assumption of gene vs sample may not be a valid one- though that is the application to which the first release will apply.

Reviewer: Sorry, I really did mean 'common'.

> Differential analysis is a frequent task in a variety of use cases. In essence, all these use cases entail taking an input matrix containing features (e.g. genes) and observations (e.g. samples), and comparing groups of observations in all or a subset of the features. The feature/ observation language here reflects our hope that this workflow will extend in future to encompass a variety of applications where an assumption of gene vs sample may not be a valid one- though that is the application to which the first release will apply.

Reviewer 2: ```suggestion
Differential analysis is a frequent task in a variety of use cases. In essence, all these use cases entail taking an input matrix containing features (e.g. genes) and observations (e.g. samples), and comparing groups of observations in all or a subset of the features. The feature/observation language here reflects our hope that this workflow will extend in future to encompass a variety of applications where an assumption of gene vs. sample may not be a valid one – though that is the application to which the first release will apply.
```

> Differential analysis is a frequent task in a variety of use cases. In essence, all these use cases entail taking an input matrix containing features (e.g. genes) and observations (e.g. samples), and comparing groups of observations in all or a subset of the features. The feature/observation language here reflects our hope that this workflow will extend in future to encompass a variety of applications where an assumption of gene vs. sample may not be a valid one – though that is the application to which the first release will apply.

Reviewer: As per my last review, I disagree with this change, please leave as it was.

--transcript_length_matrix 'salmon.merged.gene_lengths.tsv'
```

> Without the transcript lengths, for instance in earlier RNAseq workflow versions, follow the second recommendation in the [tximport documentation](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#Downstream_DGE_in_Bioconductor):

Reviewer: ```suggestion
Without the transcript lengths, for instance in earlier nf-core/rnaseq workflow versions, follow the second recommendation in the [tximport documentation](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#Downstream_DGE_in_Bioconductor):
```

lower case was to reflect the naming of the nf-core workflow- tweaked to clarify.

> "Use the tximport argument `countsFromAbundance='lengthScaledTPM'` or `'scaledTPM'`, then employ the gene-level count matrix `txi$counts` directly in downstream software, a method we call 'bias corrected counts without an offset'"

> This aligns with the **gene_counts_length_scaled.tsv** or **gene_counts_scaled.tsv** matrices in the RNAseq workflow.

Reviewer: ```suggestion
This aligns with the **gene_counts_length_scaled.tsv** or **gene_counts_scaled.tsv** matrices in the nf-core/rnaseq workflow.
```

You can optionally supply:

- `blocking` - semicolon-delimited, any additional variables (also observation columns) that should be modelled alongside the contrast variable

> - `exclude_samples_col` and `exclude_samples_values` - the former being a valid column in the samples sheet, the latter a semicolon-delimited list of values in that column which should be used to select samples prior to differential modelling. This is helpful where certain samples need to be excluded prior to analysis of a given contrast.

Reviewer 2: ```suggestion
- `exclude_samples_col` and `exclude_samples_values` - the former being a valid column in the samplesheet, the latter a semicolon-delimited list of values in that column which should be used to select samples prior to differential modelling. This is helpful where certain samples need to be excluded prior to analysis of a given contrast.
```
Normally, we call it "samplesheet". Unless you have called it "samples sheet" consistently in this pipeline?

The pipeline produces an R markdown file which, if you're proficient in R, you can use to tweak the report after it's generated (**note**- if you need the same customisations repeatedly we would recommend you supply your own template using the `report_file` parameter).

> To work with R markdown files you will need RStudio. You will also need to have the ShinyNGS R module [installed](https://github.com/pinin4fjords/shinyngs#installation), since it supplies a lot of the accessory plotting functions etc that you will need. The exact way you will do this may depend on your exact systems, but for example

Reviewer 2: ```suggestion
To work with R markdown files you will need RStudio. You will also need to have the ShinyNGS R module [installed](https://github.com/pinin4fjords/shinyngs#installation), since it supplies a lot of the accessory plotting functions etc. that you will need. The exact way to set this up may depend on your specific system, but is likely similar to the following example:
```

> To work with R markdown files you will need RStudio. You will also need to have the ShinyNGS R module [installed](https://github.com/pinin4fjords/shinyngs#installation), since it supplies a lot of the accessory plotting functions etc. that you will need. The exact way to set this up may depend on your specific system, but is likely similar to the following example:

Reviewer: Sorry, I'm happy with the original wording.

Reviewer: Please don't resolve without addressing the comments

Reviewer 3: > Please don't resolve without addressing the comments

Hi. Please check. I just kept the typoes (e.g. changing "analyis" to "analysis"), and I changed all my previous edits, due to your comment. If I missed something, please let me know.
